src/components/NucleotideEntropy.tsx

```diff
@@ -0,0 +1,706 @@
+                dot={false}
+                stroke={gene.color}
+                isAnimationActive={false}
+                key={gene.value}
+                legendType='none'
+              />
+            ))}
+          </LineChart>
+        </ResponsiveContainer>
+      </>
+    );
+  }
+};
+
+const CalculateEntropy = (
+  muts: MutationProportionEntry[] | undefined,
+  sequenceType: SequenceType,
+  deletions: boolean,
+  empty: boolean
+): PositionEntropy[] => {
+  let positionProps = new Array<PositionProportion>();
+
+  //prefill Arrays for every position
+  if (sequenceType === "nuc" && empty) {
+    positionProps = Array.apply(null, Array<PositionProportion>(29903)).map(function (x, i) {
+      let p: PositionProportion = { position: i.toString(), mutation: undefined, original: jsonRefData.nucSeq.substring(i, i+1), proportion: 0 };
+      return p;
+    });
+  } else if (sequenceType === "aa" && empty){
+    jsonRefData.genes.forEach(g => g.aaSeq.split("").forEach(function (aa, i) {
+      let p: PositionProportion = { position: g.name + ":" + aa + (i+1).toString(), mutation: undefined, original: aa, proportion: 0 };
+      positionProps.push(p);
+    }))
+  }
+
+  //map mutation proportions to positions
+  muts?.forEach(mut => {
+    if (sequenceType === 'aa') {
+      let decoded = decodeAAMutation(mut.mutation);
+      if (decoded.mutatedBase !== '-' || (decoded.mutatedBase == '-' && deletions)) {
+        let pp: PositionProportion = {
+          position: decoded.gene + ':' + decoded.originalBase + decoded.position,
+          mutation: decoded.mutatedBase,
+          original: decoded.originalBase,
+          proportion: mut.proportion,
+        };
+        positionProps.push(pp);
+      }
+    } else {
+      let decoded = decodeNucMutation(mut.mutation);
+      if (decoded.mutatedBase !== '-' || (decoded.mutatedBase == '-' && deletions)) {
+        let pp: PositionProportion = {
+          position: decoded.position.toString(),
+          mutation: decoded.mutatedBase,
+          original: decoded.originalBase,
+          proportion: mut.proportion,
+        };
+        positionProps.push(pp);
+      }
+    }
+  });
+
+  //group proportions by position
+  const positionGroups = Object.entries(groupBy(positionProps, p => p.position)).map(p => {
+    return {
+      position: p[0],
+      original: p[1][0].original,
+      proportions: p[1],
+      entropy: 0,
+    };
+  });
+
+  //calculate remaining original proportion for each position
+  positionGroups.forEach(pos => {
+    const remainder = 1 - pos.proportions.map(p => p.proportion).reduce((x, a) => x + a, 0);
+    if (remainder !== 0) {
+      pos.proportions.push({
+        position: pos.position,
+        mutation: pos.original + ' (ref)',
+        original: pos.original,
+        proportion: remainder,
+      });
+    }
+    pos.proportions = pos.proportions.filter(p => p.proportion > 0);
+  });
+
+
+  //convert proportion to entropy
+  positionGroups.map(p => {
```

Comment:
Either use `forEach` or return something from the array function in `map`, please.